### PR TITLE
Add automatic state encryption

### PR DESCRIPTION
**Pull request by yaron2**
This PR adds the automatic state encryption docs for https://github.com/dapr/dapr/issues/1090.

### CODE REVIEW

**Comment by orizohar** [comment]
We should add mentioning and linking to this new page from the security concept page and the state store overview page

**Comment by yaron2** [comment]
Added state store overview page ref.

File: daprdocs/content/en/developing-applications/building-blocks/state-management/howto-encrypt-state.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+Application state often needs to get encrypted at rest to provide stonger security in enterprise workloads or regulated environments. Dapr offers automatic client side encryption based on [AES256](https://en.wikipedia.org/wiki/Advanced_Encryption_Standard).
+
+In addition to automatic encryption, Dapr supports primary and secondary encryption keys to make it easier for developers and ops teams to enable a key rotation strategy.
+This feature is supported by all Dapr state stores.
+
+The encryption keys are fetched from a secret, and cannot be supplied as plaintext values on the `metadata` section.
+
+## Enabling automatic encryption
+
+1. enable the state encryption preview feature using a standard Dapr Configuration:
```
Comment (AaronCrawfis) [inline daprdocs/content/en/developing-applications/building-blocks/state-management/howto-encrypt-state.md]:
```suggestion
1. Enable the state encryption preview feature using a standard [Dapr Configuration]({{< ref configuration-overview.md >}}):
```

File: daprdocs/content/en/developing-applications/building-blocks/state-management/howto-encrypt-state.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
+---
+type: docs
+title: "How-To: Encrypt application state"
+linkTitle: "How-To: Encrypt application state"
```
Comment (AaronCrawfis) [inline daprdocs/content/en/developing-applications/building-blocks/state-management/howto-encrypt-state.md]:
```suggestion
title: "How-To: Encrypt application state"
linkTitle: "How-To: Encrypt state"
```

**Review by AaronCrawfis — APPROVED** [review]

**Pull request merged**